### PR TITLE
Dev 2205 deactivate contributions for early adopters

### PR DESCRIPTION
**Pull request by harsha09**
#### What's this PR do?
Limit access to Contributions dashboard for selected users by allowing access to everyone.

#### Why are we doing this? How does it help us?
Need a mechanism to disable Contributions dashboard access for few Orgs / RP users but should be accessible to all other users.

#### How should this be manually tested? Please include detailed step-by-step instructions.
To test this feature a waffle flag `spa-contributions-section-deny` need to be created using Django Admin.

How to create Waffle flag? Login to Django Admin portal -> Flags Section -> Add
Flag name should be `spa-contributions-section-deny`
1. Can enable this flag to everyone (No user should be able to access contributions dashboard).
2. Can also enable this flag by specifying users where the specified users will not be able to access the contributions dashboard.
3. `spa-contributions-section-deny` will take precedence over `spa-contributions-section-access`, you can play around with the combinations.

After creating the Waffle Flag login to Contributions with the user for whom `spa-contributions-section-deny` is enabled, Contributions tab should not be visible and navigation to /contributions redirects to 404 page.

Users for whom `spa-contributions-section-deny` is disabled and `spa-contributions-section-access` is enabled Contributions tab should be visible and be able to list contributions.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
No changes to UI but enabling and disabling Contributions for few set of users.

#### Have automated unit tests been added? If not, why?
Yes

#### How should this change be communicated to end users?
Probably user manual.

#### Are there any smells or added technical debt to note?
NA

#### Has this been documented? If so, where?
Not yet

#### What are the relevant tickets? Add a link to any relevant ones.
[DEV-2205]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
NA

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
NA

[DEV-2205]: https://news-revenue-hub.atlassian.net/browse/DEV-2205?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: spa/cypress/integration/11 - dashboard.spec.js [hunk 1]
```diff
@@ -5,32 +5,43 @@ import { DASHBOARD_SLUG, DONATIONS_SLUG, CONTENT_SLUG } from 'routes';
...
 const contentSectionFlag = {
   id: '5678',
   name: CONTENT_SECTION_ACCESS_FLAG_NAME
 };
 
-const hubAdminWithContributionsFlag = {
+const hubAdminWithContributionsAccessFlag = {
   ...hubAdminWithoutFlags,
-  flags: [{ ...contribSectionsFlag }]
+  flags: [{ ...contribSectionsAccessFlag }]
```
Comment (GuiMend) [inline spa/cypress/integration/11 - dashboard.spec.js:29]: No need to deconstruct object into a new object
```suggestion
  flags: [contribSectionsAccessFlag]
```

File: spa/cypress/integration/11 - dashboard.spec.js [hunk 2]
```diff
@@ -5,32 +5,43 @@ import { DASHBOARD_SLUG, DONATIONS_SLUG, CONTENT_SLUG } from 'routes';
...
+};
+
+const hubAdminWithContributionsDenyFlag = {
+  ...hubAdminWithoutFlags,
+  flags: [{ ...contribSectionsDenyFlag }, { ...contribSectionsAccessFlag }]
```
Comment (GuiMend) [inline spa/cypress/integration/11 - dashboard.spec.js:34]: Here too
```suggestion
  flags: [contribSectionsDenyFlag, contribSectionsAccessFlag]
```

File: spa/cypress/integration/11 - dashboard.spec.js [hunk 3]
```diff
@@ -5,32 +5,43 @@ import { DASHBOARD_SLUG, DONATIONS_SLUG, CONTENT_SLUG } from 'routes';
...
 const hubAdminWithContentFlag = {
   ...hubAdminWithoutFlags,
   flags: [{ ...contentSectionFlag }]
 };
 
-const hubAdminWithAllFlags = {
+const hubAdminWithAllAccessFlags = {
   ...hubAdminWithoutFlags,
-  flags: [{ ...contentSectionFlag }, { ...contribSectionsFlag }]
+  flags: [{ ...contentSectionFlag }, { ...contribSectionsAccessFlag }]
```
Comment (GuiMend) [inline spa/cypress/integration/11 - dashboard.spec.js:44]: here again

File: spa/src/components/dashboard/sidebar/DashboardSidebar.test.js [hunk 4]
```diff
@@ -14,6 +14,11 @@ jest.mock('utilities/flagIsActiveForUser', () => ({
   default: () => true
 }));
 
+jest.mock('utilities/hasContributionsDashboardAcessToUser', () => ({
+  __esModule: true,
+  default: () => true
+}));
+
 it('should have expected appearance and links', () => {
```
Comment (GuiMend) [inline spa/src/components/dashboard/sidebar/DashboardSidebar.test.js:22]: You could also add a unit test to check that the navigation is hidding/showing as expected. It's better to have this test as a Unit test than a cypress test (it runs quicker and checks the same thing)

File: spa/src/utilities/hasContributionsDashboardAcessToUser.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import {
+  CONTRIBUTIONS_SECTION_ACCESS_FLAG_NAME,
+  CONTRIBUTIONS_SECTION_DENY_FLAG_NAME
+} from 'constants/featureFlagConstants';
+import flagIsActiveForUser from './flagIsActiveForUser';
+
+function hasContributionsDashboardAcessToUser(featureFlags) {
```
Comment (GuiMend) [inline spa/src/utilities/hasContributionsDashboardAcessToUser.js:7]: Could also create unit tests for this function

**Review by GuiMend — CHANGES_REQUESTED** [review]
Minor comments + suggestion to add more tests to cover the changes

File: spa/cypress/integration/11 - dashboard.spec.js (hunk 1 above)
Comment (harsha09) [reply spa/cypress/integration/11 - dashboard.spec.js]: Addressed here 3beadee40bae770316d5ca5dc8e0e2a066e53da8

File: spa/cypress/integration/11 - dashboard.spec.js (hunk 3 above)
Comment (harsha09) [reply spa/cypress/integration/11 - dashboard.spec.js]: Addressed 3beadee40bae770316d5ca5dc8e0e2a066e53da8

File: spa/src/components/dashboard/sidebar/DashboardSidebar.test.js (hunk 4 above)
Comment (harsha09) [reply spa/src/components/dashboard/sidebar/DashboardSidebar.test.js:22]: Added the test using 55e6666e6897992a368530c19a0579e94f705900

File: spa/src/utilities/hasContributionsDashboardAcessToUser.js (hunk 5 above)
Comment (harsha09) [reply spa/src/utilities/hasContributionsDashboardAcessToUser.js:7]: Added tests using 8d415556506828acd455a6663a0cc96e2a3fa979

**Review by GuiMend — APPROVED** [review]
Code LGTM 👍 
Thanks for adding the additional tests @harsha09

**Review by x110dc — APPROVED** [review]

**Pull request merged**